pkg/collect/ssl_cert_incluster_collector.go

```diff
@@ -0,0 +1,216 @@
+package collect
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"log"
+	"path/filepath"
+	"time"
+
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+type CollectInClusterSSLCertInfo struct {
+	Collector    *troubleshootv1beta2.InClusterSSLCertInfo
+	BundlePath   string
+	Namespace    string
+	ClientConfig *rest.Config
+	Client       kubernetes.Interface
+	Context      context.Context
+	RBACErrors
+}
+
+// SSL Certificate Struct
+type sslCert struct {
+	CertName         string       `json:"Certificate Name"`
+	DNSNames         []string     `json:"DNS Names"`
+	IssuerCommonName string       `json:"Issuer"`
+	Organizations    []string     `json:"Issuer Organizations"`
+	CertDate         time.Time    `json:"Certificate Expiration Date"`
+	IsValid          bool         `json:"IsValid"`
+	Location         CertLocation `json:"Location,omitempty"`
+}
+
+// SSL Cert Location Struct
+type CertLocation struct {
+	Secret          string `json:"Secret Name,omitempty"`
+	SecretNamespace string `json:"Secret Namespace,omitempty"`
+}
+
+func (c *CollectInClusterSSLCertInfo) Title() string {
+	return getCollectorName(c)
+}
+
+func (c *CollectInClusterSSLCertInfo) IsExcluded() (bool, error) {
+	return isExcluded(c.Collector.Exclude)
+}
+
+func (c *CollectInClusterSSLCertInfo) Collect(progressChan chan<- interface{}) (CollectorResult, error) {
+	/* Go Client Config -- start
+	client, err := kubernetes.NewForConfig(c.ClientConfig)
+	if err != nil {
+		return nil, err
+	} */ //Go Client Config -- end
+
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+	// uses the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+	// creates the clientsets
+	client, err := kubernetes.NewForConfig(config)
```

Comment:
I think we need to use the member `Client` instance to access to the cluster (line number 27) instead of creating a new one

Reply:
ack